File: app/src/main/java/de/garten/training/depotflow/data/db/converter/SyncStatusConverter.java
package de.garten.training.depotflow.data.db.converter;

import org.greenrobot.greendao.converter.PropertyConverter;

import de.garten.training.depotflow.domain.SyncStatus;

public class SyncStatusConverter implements PropertyConverter<SyncStatus, String> {

    @Override
    public SyncStatus convertToEntityProperty(String databaseValue) {
        return SyncStatus.fromDatabaseValue(databaseValue);
    }

    @Override
    public String convertToDatabaseValue(SyncStatus entityProperty) {
        if (entityProperty == null) {
            return SyncStatus.CLEAN.toDatabaseValue();
        }
        return entityProperty.toDatabaseValue();
    }
}

File: app/src/main/java/de/garten/training/depotflow/data/db/converter/SyncStatusConverter.kt
package de.garten.training.depotflow.data.db.converter

import de.garten.training.depotflow.domain.SyncStatus
import org.greenrobot.greendao.converter.PropertyConverter

class SyncStatusConverter : PropertyConverter<SyncStatus?, String?> {
    override fun convertToEntityProperty(databaseValue: String?): SyncStatus? {
        return SyncStatus.fromDatabaseValue(databaseValue)
    }

    override fun convertToDatabaseValue(entityProperty: SyncStatus?): String? {
        if (entityProperty == null) {
            return SyncStatus.CLEAN.toDatabaseValue()
        }
        return entityProperty.toDatabaseValue()
    }
}
